docker/docker_image.go

```diff
@@ -54,6 +54,23 @@ func (i *dockerImage) Manifest() ([]byte, error) {
 	return i.cachedManifest, nil
 }
 
+// ManifestMatchesDigest returns true iff the image's manifest matches expectedDigest.
+// Error may be set if this returns false.
+func (i *dockerImage) ManifestMatchesDigest(expectedDigest string) (bool, error) {
+	// This should eventually support various digest types.
+	manifest, err := i.Manifest()
+	if err != nil {
+		return false, err
+	}
+	actualDigest, err := utils.ManifestDigest(manifest)
+	if err != nil {
+		return false, err
+	}
+	// Note that this is not doing ConstantTimeCompare; by the time we get here, the cryptographic signature must already have been verified,
+	// or we are not using a cryptographic channel and the attacker can modify the digest along with the manifest blob.
```

Comment:
can you put this comment in the func comment above? so that godoc is aware of this and users also